### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by alexfertel** [alexfertel/rust-algorithms#8]
We want to document each class of algorithms. This comes in two flavors:

* Adding README files to each directory holding a class of algorithms, for example, `sorting`.
* Adding documentation comments to each of the algorithms.

You can find examples of documentation comments [here](https://github.com/alexfertel/rust-algorithms/blob/main/src/dynamic_programming/coin_problem.rs).

General guidelines for the explanations are:
- Be concise. Keep in mind these algorithms are meant for educational purposes.
- Link to the bibliography.
- Images are better than no images.
- If it is possible, include time and space complexities.

**Comment by robbespo00** [comment]
Do you still need help for the documentation of the algorithms?

**Comment by alexfertel** [comment]
> Do you still need help for the documentation of the algorithms?

Yes, we do! You can just choose whatever algo you want and document it the best you can.

**Comment by 0xVitalii** [comment]
As I see many folders (string, sorting for example) have already had README file in their directory. Are they good or we still able to revise and improve them?

**Comment by alexfertel** [comment]
> As I see many folders (string, sorting for example) have already had README file in their directory. Are they good or we still able to revise and improve them?

Yes! Please feel free to submit improvements wherever you may see fit! @tvcapital

**Comment by patilniraj8** [comment]
Is there still any help needed?
I would like to contribute u can contact me on nirajpatil849@gmail.com

**Comment by alexfertel** [comment]
> Is there still any help needed?

Yes! Feel free to open PRs adding docs wherever you may see fit! 👍🏻 @patilniraj8

**Comment by patilniraj8** [comment]
Ok I would like to work on it thanks

**Comment by patilniraj8** [comment]
Can u  please review the pull request i created just now

**Comment by jackattack825** [comment]
I have written up ReadMe.md's for the General and Graphs folders. Tomorrow I plan to write one for the Math's folder and submit a PR.

**Comment by aditya-nv-06** [comment]
Worked on the readme.md for bit manipulation. Can you review my PR #72 in addition of I made some changes for readability.

**Comment by Peponks9** [comment]
Been recently working with a Merkle Tree. Would it be something interested to add to the repo?

**Comment by alexfertel** [comment]
Yeah, @Peponks9, feel free to open a PR!
